aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageUploadTest.java

```diff
@@ -0,0 +1,224 @@
+        // Randomly write objects to upload
+        largeFile = createTempFile(largeFileName, LARGE_FILE_SIZE);
+        smallFile = createTempFile(smallFileName, SMALL_FILE_SIZE);
+
+        // Always interact with PUBLIC access for consistency
+        options = StorageUploadFileOptions.builder()
+                .accessLevel(DEFAULT_ACCESS_LEVEL)
+                .build();
+
+        largeFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, largeFileName);
+        smallFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, smallFileName);
+    }
+
+    /**
+     * Cleans up the S3 bucket for files that were uploaded
+     * during testing processes.
+     */
+    @After
+    public void cleanUp() {
+        cleanUpS3Object(largeFileKey);
+        cleanUpS3Object(smallFileKey);
+    }
+
+    /**
+     * Tests that small file (single-part) uploads successfully.
+     *
+     * @throws StorageException if upload fails
+     */
+    @Test
+    public void testUploadSmallFile() throws StorageException {
+        StorageUploadFileResult result = Await.<StorageUploadFileResult, StorageException>result(
+                (onResult, onError) ->
+                        Amplify.Storage.uploadFile(
+                                smallFileName,
+                                smallFile.getAbsolutePath(),
+                                onResult,
+                                onError
+                        )
+        );
+        assertEquals(smallFileName, result.getKey());
+        assertS3ObjectExists(smallFileKey);
+    }
+
+    /**
+     * Tests that large file (multi-part) uploads successfully.
+     *
+     * @throws Exception if upload fails
+     */
+    @Test
+    public void testUploadLargeFile() throws Exception {
+        final CountDownLatch completed = new CountDownLatch(1);
+        Amplify.Storage.uploadFile(
+                largeFileName,
+                largeFile.getAbsolutePath(),
+                onResult -> {
+                    assertEquals(largeFileName, onResult.getKey());
+                    assertS3ObjectExists(largeFileKey);
+                    completed.countDown();
+                },
+                onError -> fail("Upload was not successful.")
+        );
+        completed.await(DEFAULT_TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Tests that file upload operation can be canceled while the
+     * transfer hasn't completed yet.
+     *
+     * @throws Exception if upload is not canceled successfully
+     *         before timeout
+     */
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testUploadFileIsCancelable() throws Exception {
+        final CountDownLatch canceled = new CountDownLatch(1);
+
+        // Begin uploading large file
+        StorageUploadFileOperation<?> op = Amplify.Storage.uploadFile(
+                largeFileName,
+                largeFile.getAbsolutePath(),
+                options,
+                onResult -> fail("Upload finished before being successfully cancelled."),
```

Comment:
Is there onProgress on Amplify.Storage.download|upload’s?

Reply:
There isn't at the moment. Progress will only be accessible via Hub in Android

Reply:
Speaking of which, it might be a good idea to add progress directly to the operation? I'll look into that idea.